scene/resources/skeleton_modification_3d_jiggle.h

```diff
@@ -0,0 +1,137 @@
+/* a copy of this software and associated documentation files (the       */
+/* "Software"), to deal in the Software without restriction, including   */
+/* without limitation the rights to use, copy, modify, merge, publish,   */
+/* distribute, sublicense, and/or sell copies of the Software, and to    */
+/* permit persons to whom the Software is furnished to do so, subject to */
+/* the following conditions:                                             */
+/*                                                                       */
+/* The above copyright notice and this permission notice shall be        */
+/* included in all copies or substantial portions of the Software.       */
+/*                                                                       */
+/* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,       */
+/* EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF    */
+/* MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.*/
+/* IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY  */
+/* CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,  */
+/* TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE     */
+/* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                */
+/*************************************************************************/
+
+#include "scene/3d/skeleton_3d.h"
+#include "scene/resources/skeleton_modification_3d.h"
+
+#ifndef SKELETONMODIFICATION3DJIGGLE_H
+#define SKELETONMODIFICATION3DJIGGLE_H
+
+class SkeletonModification3DJiggle : public SkeletonModification3D {
+	GDCLASS(SkeletonModification3DJiggle, SkeletonModification3D);
+
+private:
+	struct Jiggle_Joint_Data {
+		String bone_name = "";
+		int bone_idx = -1;
+
+		bool override_defaults = false;
+		float stiffness = 3;
+		float mass = 0.75;
+		float damping = 0.75;
+		bool use_gravity = false;
+		Vector3 gravity = Vector3(0, -6.0, 0);
+		float roll = 0;
+
+		Vector3 cached_rotation = Vector3(0, 0, 0);
+		Vector3 force = Vector3(0, 0, 0);
+		Vector3 acceleration = Vector3(0, 0, 0);
+		Vector3 velocity = Vector3(0, 0, 0);
+		Vector3 last_position = Vector3(0, 0, 0);
+		Vector3 dynamic_position = Vector3(0, 0, 0);
+
+		Vector3 last_noncollision_position = Vector3(0, 0, 0);
+	};
+
+	NodePath target_node;
+	ObjectID target_node_cache;
+	Vector<Jiggle_Joint_Data> jiggle_data_chain;
+
+	float stiffness = 3;
+	float mass = 0.75;
+	float damping = 0.75;
+	bool use_gravity = false;
+	Vector3 gravity = Vector3(0, -6.0, 0);
+
+	bool use_colliders = false;
+	uint32_t collision_mask = 1;
+
+	void update_cache();
+	void _execute_jiggle_joint(int p_joint_idx, Node3D *p_target, float p_delta);
+	void _update_jiggle_joint_data();
+
+protected:
+	static void _bind_methods();
+	bool _get(const StringName &p_path, Variant &r_ret) const;
+	bool _set(const StringName &p_path, const Variant &p_value);
+	void _get_property_list(List<PropertyInfo> *p_list) const;
```

Comment:
Please consider to add `virtual` and `override`, this will allow the compiler to better guide us in case of refactoring:
```c++
	virtual bool _get(const StringName &p_path, Variant &r_ret) const override;
	virtual bool _set(const StringName &p_path, const Variant &p_value) override;
	virtual void _get_property_list(List<PropertyInfo> *p_list) const override;
```